test/drivers/test_cqrrp.cc

```diff
@@ -227,49 +290,103 @@ TEST_F(TestCQRRP, CQRRP_blocked_low_rank) {
-TEST_F(TestCQRRP, something) {
-    int64_t m = 10;
-    int64_t n = 5;
+TEST_F(TestCQRRP, CQRRP_blocked_near_zero_luqr) {
+    int64_t m = 1000;//5000;
+    int64_t n = 1000;//2000;
+    int64_t k = 1000;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 100;//500;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
     auto state = RandBLAS::RNGState();
 
-    std::vector<double> A(m * n, 0.0);
-    std::vector<double> B(m * n, 0.0);
-    std::vector<double> C(m * 2 * n, 0.0);
-    std::vector<double> D(m * n, 0.0);
-    std::vector<double> D_cpy(m * n, 0.0);
-    std::vector<double> D_space(m * n, 0.0);
+    CQRRPTestData<double> all_data(m, n, k);
+    RandLAPACK::CQRRP_blocked<double, r123::Philox4x32> CQRRP_blocked(true, tol, b_sz);
+    CQRRP_blocked.nnz = 2;
+    CQRRP_blocked.num_threads = 4;
 
-    std::vector<double> tau(n * 2, 0.0);
+    std::fill(&(all_data.A.data())[0], &(all_data.A.data())[m * n], 0.0);
+    all_data.A[1000*200 + 10] = 1;
 
-    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
-    RandLAPACK::gen::mat_gen(m_info, A.data(), state);
-    RandLAPACK::gen::mat_gen(m_info, B.data(), state);
-    RandLAPACK::gen::mat_gen(m_info, D.data(), state);
-    lapack::lacpy(MatrixType::General, m, n, D.data(), m, D_cpy.data(), m);
+    norm_and_copy_computational_helper(norm_A, all_data);
+#if !defined(__APPLE__)
+    test_CQRRP_general(d_factor, norm_A, all_data, CQRRP_blocked, state);
+#endif
+}
 
-    lapack::geqrf(m, n, A.data(), m, tau.data());
-    lapack::geqrf(m, n, B.data(), m, tau.data() + n);
+// Note: If Subprocess killed exception -> reload vscode
+TEST_F(TestCQRRP, CQRRP_blocked_half_zero_luqr) {
+    int64_t m = 5000;//5000;
+    int64_t n = 2000;//2000;
+    int64_t k = 2000;
+    double d_factor = 1;//1.0;
+    int64_t b_sz = 500;//500;
+    double norm_A = 0;
+    double tol = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state = RandBLAS::RNGState();
 
-    // Method 1
-    lapack::lacpy(MatrixType::Lower, m, n, A.data(), m, C.data(), m);
-    lapack::lacpy(MatrixType::Lower, m, n, B.data(), m, C.data() + (m * n), m);
-    lapack::ormqr(Side::Left, Op::NoTrans, m, n, m, C.data(), m, tau.data(), D.data(), m);
+    CQRRPTestData<double> all_data(m, n, k);
+    RandLAPACK::CQRRP_blocked<double, r123::Philox4x32> CQRRP_blocked(true, tol, b_sz);
+    CQRRP_blocked.nnz = 2;
+    CQRRP_blocked.num_threads = 4;
 
-    char name [] = "D through ormqr";
-    RandBLAS::util::print_colmaj(m, n, D.data(), name);
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    RandLAPACK::gen::mat_gen(m_info, all_data.A.data(), state);
+    std::fill(&(all_data.A.data())[m * n / 2], &(all_data.A.data())[m * n], 0.0);
+
+    norm_and_copy_computational_helper(norm_A, all_data);
+#if !defined(__APPLE__)
+    test_CQRRP_general(d_factor, norm_A, all_data, CQRRP_blocked, state);
+#endif
+}
```

Comment:
As before, the entire test fixture should be wrapped with ``#if !defined(__APPLE__)``

Reply:
Fixed.